scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/clazz/ClassDefinitionDiscovery.scala

```diff
@@ -10,7 +10,8 @@ import pl.touk.nussknacker.engine.variables.MetaVariables
 
 import scala.collection.mutable
 
-object ClassDefinitionDiscovery extends LazyLogging with ExecutionTimeMeasuring {
+class ClassDefinitionDiscovery(settings: ClassExtractionSettings) extends LazyLogging with ExecutionTimeMeasuring {
+  private val classDefinitionExtractor = new ClassDefinitionExtractor(settings)
```

Comment:
Let's move `ClassDefinitionExtractor` to constructor parameters. Thanks to that we will be closer to keeping DI in one place - on top.

Reply:
Done